src/widgetastic/ouia.py

```diff
@@ -0,0 +1,83 @@
+from widgetastic.utils import ParametrizedLocator
+from widgetastic.xpath import quote
+from widgetastic.widget.base import WTMixin
+from widgetastic.widget.base import Widget
+from widgetastic.widget.base import View
+from widgetastic.widget.base import ClickableMixin
+
+
+class OUIABase:
+    """
+    Base class for ``OUIA`` support. According to the spec ``OUIA`` compatible components may have
+    the following attributes in the root level HTML element:
+        * data-ouia-component-type
+        * data-ouia-component-id
+        * data-ouia-safe
+    https://ouia.readthedocs.io/en/latest/README.html#ouia-component
+    """
+
+    ROOT = ParametrizedLocator(".//*[@data-ouia-component-type={@component_type}{@component_id}]")
+
+    def __init__(self, component_type, component_id=None, namespace=None):
+        component_type = f"{namespace}/{component_type}" if namespace else component_type
+        self.component_type = quote(component_type)
+        component_id = f" and @data-ouia-component-id={quote(component_id)}" if component_id else ""
+        self.component_id = component_id
+        self.locator = self.ROOT.locator
+
+    @property
+    def is_safe(self):
+        """
+        An attribute called data-ouia-safe, which is True only when the component is in a static
+        state, i.e. no animations are occurring. At all other times, this value MUST be False.
+        """
+        return "true" in self.browser.get_attribute("data-ouia-safe", self)
+
+
+class OUIAMixin(WTMixin, OUIABase):
+    """A mixin for Widgets and Views for adding ``OUIA`` support.
+
+    It might be useful when you need to add ``OUIA`` support into existing widgets and view and
+    preserve backward compatibility. See https://github.com/RedHatQE/widgetastic.patternfly4/blob/master/src/widgetastic_patternfly4/ouia.py  # noqa
+    """
+
+    pass
+
+
+class OUIAGenericView(View, OUIAMixin):
+    """A base class for any OUIA compatible view.
+
+    Children classes must have the same name as the value of ``data-ouia-component-type`` attriburte
+    of the root HTML element. Besides children classes should define ``OUIA_NAMESPACE`` attribute if
+    it's appicable.
+
+    Args:
+        component_id: value of data-ouia-component-id attribute.
+    """
+
+    OUIA_NAMESPACE = None
+
+    def __init__(self, parent, component_id=None, logger=None, **kwargs):
+        OUIAMixin.__init__(self, type(self).__name__, component_id, namespace=self.OUIA_NAMESPACE)
```

Comment:
The only thing that should be here is a super call. 
The moment we manually integrate multiple base class ctors, we set up something horrible.

Reply:
@RonnyPfannschmidt What should I do with the arguments? `Widget` and `OUIABase` have different `__init__` signatures. Should I just copy and paste `OUIABase` methods and attrs into `OUIAGenericView` and `OUIAGenericWidget` for sake of simplicity?

Reply:
I couldn't figure out anything better than:
```python
super().set_ouia_attrs(type(self).__name__, component_id, namespace=self.OUIA_NAMESPACE)
super().__init__(parent, logger=logger, **kwargs)
```

Reply:
The eventual correct solution will require cooperative ctors on all involved classes

Reply:
I don't see how to do that without patching `Widget` and `View` classes. I think this shouldn't be done in this PR. Another option would be just copying and pasting attributes and methods from `OUIABase` class.

Reply:
@RonnyPfannschmidt I tried to do that and slightly changed `Widget` and `View` classes.